apis/python/src/tiledbsoma/_multiscale_image.py

```diff
@@ -0,0 +1,257 @@
+
+        Reads the bounding box of the input region from the requested image level. This
+        will return a :class:`SpatialRead` with the image data stored as a
+        :class:`pa.Tensor`.
+
+        Args:
+            level: The image level to read the data from. May use index of the level
+                or the image name.
+            region: The region to query. May be a box in the form
+                [x_min, y_min, x_max, y_max] (for 2D images), a box in the form
+                [x_min, y_min, z_min, x_max, y_max, z_max] (for 3D images), or
+                a shapely Geometry.
+            channel_coords: An optional slice that defines the channel coordinates
+                to read.
+            region_transform: An optional coordinate transform that provides the
+                transformation from the provided region to the reference level of this
+                image. Defaults to ``None``.
+            region_coord_space: An optional coordinate space for the region being read.
+                The axis names must match the input axis names of the transform.
+                Defaults to ``None``, coordinate space will be inferred from transform.
+            result_order: the order to return results, specified as a
+                :class:`~options.ResultOrder` or its string value.
+
+        Returns:
+            The data bounding the requested region as a :class:`SpatialRead` with
+            :class:`pa.Tensor` data.
+
+        Lifecycle:
+            Experimental.
+        """
+        raise NotImplementedError()
+
+    # Metadata operations
+
+    @property
+    def axis_names(self) -> Tuple[str, ...]:
+        """The name of the image axes.
+
+        Lifecycle:
+            Experimental.
+        """
+        raise NotImplementedError()
+
+    @property
+    def coordinate_space(self) -> Optional[CoordinateSpace]:
+        """Coordinate space for this multiscale image.
+
+        Lifecycle:
+            Experimental.
+        """
+        raise NotImplementedError()
+
+    @coordinate_space.setter
+    def coordinate_space(self, value: CoordinateSpace) -> None:
+        """Coordinate space for this multiscale image.
+
+        Lifecycle:
+            Experimental.
+        """
+        raise NotImplementedError()
+
+    def get_transform_from_level(self, level: Union[int, str]) -> ScaleTransform:
+        """Returns the transformation from user requested level to image reference
+        level.
+
+        Lifecycle:
+            Experimental.
+        """
+        raise NotImplementedError()
+
+    def get_transform_to_level(self, level: Union[int, str]) -> ScaleTransform:
+        """Returns the transformation from the image reference level to the user
+        requested level.
+
+        Lifecycle:
+            Experimental.
+        """
+        raise NotImplementedError()
+
+    @property
+    def image_type(self) -> str:
+        """The order of the axes as stored in the data model.
```

Comment:
Maybe given an example return value? On this PR or a subsequent.

Reply:
I added a shortcut story for this (sc-56341).